File: backend/app/models/match.py
from pydantic import BaseModel
from datetime import datetime
from typing import Literal


class Match(BaseModel):
    id: str
    sport: str                          # "futbol", "tenis", "basquet", etc.
    competition: str                    # "Liga Profesional Argentina"
    home_team: str
    away_team: str
    home_score: int | None = None
    away_score: int | None = None
    status: Literal["live", "upcoming", "finished"]
    minute: str | None = None           # "69'" / "3er C" / "2do set"
    datetime: datetime | None = None    # fecha y hora UTC del partido
    start_time: str | None = None       # "21:30" — hora local ARG (display)
    argentina_relevance: Literal["seleccion", "club_arg", "jugador_arg", "none"] = "none"
    argentina_team: str | None = None   # equipo/jugador ARG involucrado
    broadcast: str | None = None        # "TyC Sports", "ESPN 3", etc.

File: frontend/src/types/match.ts
export type MatchStatus = 'live' | 'upcoming' | 'finished'
export type ArgentinaRelevance = 'seleccion' | 'club_arg' | 'jugador_arg' | 'none'

export interface Match {
  id: string
  sport: string
  competition: string
  home_team: string
  away_team: string
  home_score: number | null
  away_score: number | null
  status: MatchStatus
  minute: string | null
  datetime: string | null
  start_time: string | null
  argentina_relevance: ArgentinaRelevance
  argentina_team: string | null
  broadcast: string | null
}
